src/library_pthread.js

```diff
@@ -1169,8 +1171,15 @@ var LibraryPThread = {
 
 #if MODULARIZE
   $establishStackSpaceInJsModule: function(stackBase, stackMax) {
```

Comment:
stackBase always means the lowest address everywhere i think, only make needs to be renamed.

Reply:
I'd suggest renaming STACK_TOP to SP or STACK_POINTER, and STACK_MAX to STACK_LIMIT. stackBase/stackMax could then be used to refer to the lowest/highest address. Would that work ?

Reply:
> stackBase always means the lowest address everywhere i think, only make needs to be renamed.

I don't think that's true for `STACK_BASE`. Here is what hello world gets:
```
var STATIC_BASE = 1024,
    STACK_BASE = 5246688,
    STACKTOP = STACK_BASE,
    STACK_MAX = 3808,
    DYNAMIC_BASE = 5246688,
    DYNAMICTOP_PTR = 3776;
```
`STACK_BASE` is the initial value of the stack pointer, which for the wasm backend is the highest address. And as mentioned above I think it's important we keep `STACK_BASE` and `stackBase` consistent.

> I'd suggest renaming STACK_TOP to SP or STACK_POINTER, and STACK_MAX to STACK_LIMIT. stackBase/stackMax could then be used to refer to the lowest/highest address. Would that work ?

I'm not not picky about the names, but I slightly worry about expanding this PR to such a larger renaming - I think it will end up a nontrivial amount of work. In particular we may want to do the same for `DYNAMIC_*` and `STATIC_*` for consistency as they all use the same naming convention now. Another issue is that we hope to get rid of fastcomp eventually - it won't happen soon, but once it does, we'll just have one direction for the stack, so all this will become easier to refactor. So I'd prefer to leave a larger refactoring to a later PR (by you, if you're interested, of course).

However, if you'd rather do it here, then that sounds ok too. Those names sound generally ok. I like that LIMIT makes sense regardless of the direction the stack grows. Instead of `base/max` how about `min/max` for the lowest/highest addresses? And for the current location, `STACK_POINTER` is probably least confusing, but then we'd want `DYNAMIC_POINTER, STATIC_POINTER` for those other two, which I'm not sure about... but maybe we can leave those aside.